from fastapi import FastAPI, HTTPException

app = FastAPI()

items = { "foo": "The Foo Wrestlers" }

@app.get("/items/{item_id}")
async def read_item(item_id: str):      # Função de operações de caminho
    if item_id not in items:
        raise HTTPException(        # Função de utilidade
            status_code=404, 
            detail="Item not found",
            headers={ "X-Error": "Deu erro doidão" }
            )       
    return { "item": items[item_id] }

# O HTTPexception irá finalizar a requisição e enviar o erro HTTP dele para o cliente (exceção)
0 comments on commit 031a9d8
@AndersonMartins1
Comment
